src/Orient/Orient.Client/API/OClient.cs

```diff
@@ -97,63 +84,24 @@ public static int DatabasePoolCurrentSize(string alias)
 
         internal static void DropAndEstablishConnectionsInAllPools()
         {
-            lock (_syncRoot)
+            foreach (var pool in _databasePools.Values)
             {
-                foreach (var pool in _databasePools)
-                {
-                    pool.DropAndEstablishAllConnections();
-                }
+                pool.DropAndEstablishAllConnections();
             }
         }
 
         internal static Connection ReleaseConnection(string alias)
         {
-            lock (_syncRoot)
-            {
-                Connection connection = null;
-
-                if (_databasePools.Exists(db => db.Alias == alias))
-                {
-                    DatabasePool pool = _databasePools.Find(db => db.Alias == alias);
-
-                    if (pool != null)
-                    {
-                        // deque free database connection if the pool has one
-                        if (pool.CurrentSize > 0)
-                        {
-                            connection = pool.DequeueConnection();
-                        }
-                        // if the pool is empty - create new dedicated database connection
-                        else if (pool.CurrentSize == 0)
-                        {
-                            connection = new Connection(pool.Hostname, pool.Port, pool.DatabaseName, pool.DatabaseType, pool.UserName, pool.UserPassword, alias, true);
-                        }
-                    }
-                }
-
-                return connection;
-            }
+            return _databasePools[alias].DequeueConnection();            

```

Comment:
If the alias does not exist `KeyNotFoundException` will be thrown, which is much better than returning with `null` and getting a NPE later (like it was implemented before)